Source/Widgets/Cesium3DTilesInspector/Cesium3DTilesInspectorViewModel.js

```diff
@@ -0,0 +1,880 @@
+                    return;
+                }
+                var currentFeature = this._feature;
+                if (defined(currentFeature) && defined(currentFeature._batchTable) && !currentFeature._batchTable.isDestroyed()) {
+                    // Restore original color to feature that is no longer selected
+                    var frameState = this._scene.frameState;
+                    if (!this.colorize && defined(this._style)) {
+                        currentFeature.color = this._style.color.evaluateColor(frameState, currentFeature, scratchColor);
+                    } else {
+                        currentFeature.color = oldColor;
+                    }
+                }
+                if (defined(feature)) {
+                    // Highlight new feature
+                    Color.clone(feature.color, oldColor);
+                    feature.color = highlightColor;
+                }
+                this._feature = feature;
+            }
+        }
+    });
+
+    /**
+     * Toggles the pick tileset mode
+     */
+    Cesium3DTilesInspectorViewModel.prototype.togglePickTileset = function() {
+        this.pickActive = !this.pickActive;
+    };
+
+    /**
+     * Toggles the inspector visibility
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleInspector = function() {
+        this.inspectorVisible = !this.inspectorVisible;
+    };
+
+    /**
+     * Toggles the visibility of the tileset section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleTileset = function() {
+        this.tilesetVisible = !this.tilesetVisible;
+    };
+
+    /**
+     * Toggles the visibility of the display section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleDisplay = function() {
+        this.displayVisible = !this.displayVisible;
+    };
+
+    /**
+     * Toggles the visibility of the update section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleUpdate = function() {
+        this.updateVisible = !this.updateVisible;
+    };
+
+    /**
+     * Toggles the visibility of the logging section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleLogging = function() {
+        this.loggingVisible = !this.loggingVisible;
+    };
+
+    /**
+     * Toggles the visibility of the style section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleStyle = function() {
+        this.styleVisible = !this.styleVisible;
+    };
+
+    /**
+     * Trims tile cache
+     */
+    Cesium3DTilesInspectorViewModel.prototype.trimTilesCache = function() {
+        if (defined(this._tileset)) {
+            this._tileset.trimLoadedTiles();
+        }
+    };
+
+    /**
+     * Compiles the style in the style editor
+     */
+    Cesium3DTilesInspectorViewModel.prototype.compileStyle = function() {
+        if (defined(this._style)) {
+            if (this.styleString !== JSON.stringify(this._style.style)) {
+                if (defined(this._tileset)) {
```

Comment:
Invert and combine some of these statements to avoid 3 levels of `if` nesting.  A bunch of `this` properties are accessed multiple times in this function, break them out into locals.